tests/Bridge/Laravel/Http/Middlewares/AuditMiddlewareTest.php

```diff
@@ -0,0 +1,259 @@
+            $this->getRequest(),
+            static function (): ResponseInterface {
+                return new TextResponse('OK');
+            }
+        );
+
+        self::assertInstanceOf(ResponseInterface::class, $response);
+        self::assertSame('OK', (string)$response->getBody());
+    }
+
+    /**
+     * Test if exception thrown by http logger service is logged using logger interface
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHttpLoggerExceptionsAreLoggedByLoggerInterface(): void
+    {
+        $httpLogger = new HttpLoggerExceptionStub(); // this stub specifically throws an exception
+        $logHandler = new LogHandlerStub();
+        $middleware = $this->getMiddleware($httpLogger, $logHandler);
+
+        $middleware->handle(
+            $this->getRequest(),
+            static function (): Response {
+                return new Response('OK');
+            }
+        );
+
+        self::assertCount(1, $logHandler->getLogs());
+        $log = $logHandler->getLogs()[0];
+        self::assertArrayHasKey('message', $log);
+    }
+
+    /**
+     * Test to assert that an exception is caught and the process
+     * continues to generate psr7 request/response
+     *
+     * @return void
+     */
+    public function testProcessingContinuesWhenExceptionIsThrownInOriginalProcess(): void
+    {
+        $httpLogger = new HttpLoggerStub();
+        $middleware = $this->getMiddleware($httpLogger);
+
+        try {
+            $middleware->handle(
+                $this->getRequest(),
+                static function (): void {
+                    throw new InvalidDocumentClassException('Something failed.');
+                }
+            );
+        } catch (Exception $exception) {
+            // catching exception so we can assert the the process continues to send data to http logger
+        }
+
+        self::assertInstanceOf(ServerRequestInterface::class, $httpLogger->getRequest());
+        self::assertNull($httpLogger->getResponse());
+        self::assertInstanceOf(DateTime::class, $httpLogger->getNow());
+    }
+
+    /**
+     * Test if exceptions thrown when trying to create psr requests are logged
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testPsrRequestExceptionsAreLoggedByLoggerInterface(): void
+    {
+        $logHandler = new LogHandlerStub();
+        $middleware = $this->getMiddleware(null, $logHandler);
+
+        $middleware->handle(
+            new Request(), // this will throw an exception while trying to create psr7 request as HOST is missing
+            static function (): Response {
+                return new Response('OK');
+            }
+        );
+
+        self::assertCount(1, $logHandler->getLogs());
+        $log = $logHandler->getLogs()[0];
+        self::assertArrayHasKey('message', $log);
```

Comment:
How do we know if we logged the right message?

Reply:
the Logger just takes in the exception and the message is $exception->getMessage(). Because I think we are not specifically throwing the exception here (we are only catching what might had been thrown in the underlying process), I am not sure we can match the exception messages here. This was just to make sure the logger is doing its thing and created the excepted array with error message. I might be wrong, but I think because I not testing logger functionality here and only concern was that the middleware sends data to logger or not.

Reply:
I see, you're testing the behaviour in createPsr7Request() (which is fine) - but right now all you know is that "Something" failed and was logged. It's pretty easy to capture the message with xdebug, and assert that value, and then if this is refactored and something else is throwing an error, we'll know.